aptos-move/framework/aptos-framework/sources/transaction_context.move

```diff
@@ -1,4 +1,43 @@
 module aptos_framework::transaction_context {
+
+    use std::features;
+
+    /// UUID feature is not supported.
+    const EUUID_NOT_SUPPORTED: u64 = 3;
+
+    /// A wrapper denoting universally unique identifer (UUID)
+    /// for storing an address
+    struct UUID has drop, store {
+        unique_address: address
+    }
+
+    /// Return the transaction hash of the current transaction
+    public native fun get_txn_hash(): vector<u8>;
+
+    /// Return a universally unique identifier (of type address) generated
+    /// by hashing the transaction hash of this transaction and a sequence number 
+    /// specific to this transaction. This function can be called any
+    /// number of times inside a single transaction. Each such call increments
+    /// the sequence number and generates a new unique address
+    native fun create_unique_address(): address;
+
+    /// Return a universally unique identifier. Internally calls
+    /// the private function `create_unique_address`. This function is
+    /// created for to feature gate the `create_unique_address` function.
+    public fun create_unique_addr(): address {
+        assert!(features::uuids_enabled(), EUUID_NOT_SUPPORTED);
+        create_unique_address()
+    }
+ 
     /// Return the script hash of the current entry function.
     public native fun get_script_hash(): vector<u8>;
+
+    /// This method runs `create_unique_address` native function and returns 
+    /// the generated unique address wrapped in the UUID class.
+    public fun create_uuid(): UUID {
```

Comment:
How is this going to be use? UUID is an internal type so other modules would not be able to see the inner type anyway.

Reply:
David Wolinsky suggested this change. The idea is that, if we only expose `create_unique_address`,  then some malicious code could actually say that it is creating a unique address but instead use some address like 0x1. If we expose, `create_uuid`, then we know that UUIDs can be generated only by `create_uuid` method and are more confident that the address is generated uniquely and randomly. Does that make sense?

Reply:
We do need to have a function in this module that given UUID gives you the unique_address it wraps

Reply:
Done.